src/plugins/intel_npu/src/al/include/npuw_private_properties.hpp

```diff
@@ -0,0 +1,245 @@
+ * Device list to try in order.
+ * Possible values: Comma-separated list of devices. E.g., "NPU,GPU,CPU".
+ * Default value: "NPU,CPU"
+ */
+static constexpr ov::Property<std::string> devices{"NPUW_DEVICES"};
+
+/**
+ * @brief
+ * Type: std::string.
+ * Force the specific subgraph to specific device. The device must be present in the NPUW_DEVICES list.
+ * Possible values: Comma-separated "Subgraph index:OpenVINO device name" pairs,
+ * e.g. "0:CPU,1:NPU".
+ * Default value: empty.
+ */
+static constexpr ov::Property<std::string> submodel_device{"NPUW_SUBMODEL_DEVICE"};
+
+namespace partitioning {
+namespace online {
+/**
+ * @brief
+ * Type: std::string.
+ * Specify which partitioning pipeline to run.
+ * Possible values: "NONE", "INIT", "JUST", "REP".
+ * Default value: "REP".
+ */
+static constexpr ov::Property<std::string> pipeline{"NPUW_ONLINE_PIPELINE"};
+
+/**
+ * @brief
+ * Type: std::string.
+ * Forbids operation(s) and/or predefined pattern(s) to compile and run on a
+ * specified device. Only compatible with online partitioning.
+ * Possible values: comma-separated list of operations slash device, e.g.
+ *                  "Op:Select/NPU,P:RMSNorm/NPU".
+ * Default value: empty.
+ */
+static constexpr ov::Property<std::string> avoid{"NPUW_ONLINE_AVOID"};
+
+/**
+ * @brief
+ * Type: std::size_t.
+ * Lower boundary of partition graph size the plugin can generate.
+ * Used to control fusion term criteria in online partitioning.
+ * WO for online partitioning.
+ * Possible values: Integer >= 10.
+ * Default value: 10.
+ */
+static constexpr ov::Property<std::size_t> min_size{"NPUW_ONLINE_MIN_SIZE"};
+
+/**
+ * @brief
+ * Type: std::string.
+ * Dump online partitioning to the specified file.
+ * This partitioning can be reused via NPUW_PLAN property later.
+ * Possible values: Path to .xml file.
+ * Default value: empty.
+ */
+static constexpr ov::Property<std::string> dump_plan{"NPUW_ONLINE_DUMP_PLAN"};
+}  // namespace online
+
+/**
+ * @brief
+ * Type: std::string.
+ * Set plan file to use by offline partitioning.
+ * Possible values: Path to .xml file.
+ * Default value: empty.
+ */
+static constexpr ov::Property<std::string> plan{"NPUW_PLAN"};
+
+/**
+ * @brief
+ * Type: bool.
+ * Perform function call folding if there are repeating blocks in the graph.
+ * Possible values: true, false.
+ * Default value: false.
+ */
+static constexpr ov::Property<bool> fold{"NPUW_FOLD"};
+
+/**
+ * @brief
+ * Type: bool.
+ * Cut-off weighs from repeating blocks, but don't do folding.
+ * Decompression cut-off may still happen. Conflicts with NPUW_FOLD.
+ * Possible values: true, false.
+ * Default value: false.
+ */
+static constexpr ov::Property<bool> cwai{"NPUW_CWAI"};
+
+/**
+ * @brief
+ * Type: std::string.
+ * Promotional data type for weights decompression. Works only with function "NPUW_FOLD"ing.
+ * Possible values: "i8", "f16"
+ * Default value: empty.
+ */
+static constexpr ov::Property<std::string> dcoff_type{"NPUW_DCOFF_TYPE"};
```

Comment:
If the value is not set, when it is `ov::element::Type`, what would be the right default value?

If it not set now, we do nothing. If it is set, we do something. I see how it can be done with string, but once the type is promoted to `ov::element::Type` - need to think some logic around that

Reply:
`ov::element::Type_t::undefined` maybe. I'm not yet familiar with the configuration flow, so I can't tell if this really is a good move.